### PR TITLE
PHPC-629: Bump libbson and libmongoc to 1.4.0-beta1

### PR DESCRIPTION
**Pull request by jmikola**
This PR incorporates several interdependent issues:

 * https://jira.mongodb.org/browse/PHPC-629
 * https://jira.mongodb.org/browse/PHPC-754
 * https://jira.mongodb.org/browse/PHPC-755
 * https://jira.mongodb.org/browse/PHPC-758

### CODE REVIEW

File: config.m4 [hunk 1]
```diff
@@ -206,7 +206,6 @@ if test "$MONGODB" != "no"; then
     bson-atomic.c \
     bson-clock.c \
     bson-context.c \
-    bson-decimal128.c \
```
Comment (derickr) [inline config.m4]: Weren't we going to leave it compiled in? AFAIK, @bjori said there was a feature flag protection in it.

File: tests/bson/bson-binary-001.phpt [hunk 2]
```diff
@@ -54,36 +54,36 @@ bool(true)
 bool(true)
 bool(true)
 bool(true)
-Test#0 { "binary" : { "$type" : "00", "$binary" : "cmFuZG9tIGJpbmFyeSBkYXRh" } }
-string(73) "{ "binary" : { "$type" : "00", "$binary" : "cmFuZG9tIGJpbmFyeSBkYXRh" } }"
-string(73) "{ "binary" : { "$type" : "00", "$binary" : "cmFuZG9tIGJpbmFyeSBkYXRh" } }"
+Test#0 { "binary" : { "$binary" : "cmFuZG9tIGJpbmFyeSBkYXRh", "$type" : "00" } }
+string(73) "{ "binary" : { "$binary" : "cmFuZG9tIGJpbmFyeSBkYXRh", "$type" : "00" } }"
+string(73) "{ "binary" : { "$binary" : "cmFuZG9tIGJpbmFyeSBkYXRh", "$type" : "00" } }"
```
Comment (derickr) [inline tests/bson/bson-binary-001.phpt]: Why are the binary tests changing?

File: config.m4 (hunk 1 above)
Comment (jmikola) [inline config.m4]: He was mistaken. Both this and the client metadata sources in libmongoc are missing an `#ifdef` wrap.

File: tests/bson/bson-binary-001.phpt (hunk 2 above)
Comment (jmikola) [inline tests/bson/bson-binary-001.phpt]: See: [CDRIVER-1340](https://jira.mongodb.org/browse/CDRIVER-1340). I referred ot that in the full commit message for these changes.

**Comment by derickr** [comment]
LGTM

**Pull request merged**